apps/api/src/app/shared/services/cache/cache.service.ts

```diff
@@ -0,0 +1,72 @@
+import Redis from 'ioredis';
+import { getRedisPrefix } from '@novu/shared';
+
+interface ICacheService {
+  set(key: string, value: string, options?: CachingConfig);
+  get(key: string);
+  del(key: string);
+  reset();
+}
+
+export type CachingConfig = {
+  ttl?: number;
+};
+
+export class CacheService {
+  private readonly client: Redis;
+  private readonly DEFAULT_TTL = 60;
+
+  constructor() {
+    this.client = new Redis(Number(process.env.REDIS_PORT), process.env.REDIS_HOST, {
+      password: process.env.REDIS_PASSWORD,
+      connectTimeout: 50000,
+      keepAlive: 30000,
+      family: 4,
+      keyPrefix: getRedisPrefix(),
+    });
+  }
+
+  public async set(key: string, value: string, options?: CachingConfig) {
+    this.client.set(key, value);
+    this.updateTtl(key, options);
+  }
+
+  public async keys(pattern = '*') {
```

Comment:
```suggestion
  const ALL_KEYS = '*';
  ...
  public async keys(pattern = ALL_KEYS) {
```
This suggestion is for letting clear to anyone not familiar with Redis what that pattern means.

Reply:
Love it! @p-fernandez , I will add it to the following branch.